enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.unit.spec.tsx

```diff
@@ -0,0 +1,68 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { Button } from "metabase/ui";
+
+import { MultiStepPopover } from "./MultiStepPopover";
+
+const setup = (currentStep = "step1") => {
+  return render(
+    <MultiStepPopover currentStep={currentStep}>
+      <MultiStepPopover.Target>
+        <Button>Hello world</Button>
+      </MultiStepPopover.Target>
+      <MultiStepPopover.Step value="step1">
+        Step 1 content
+      </MultiStepPopover.Step>
+      <MultiStepPopover.Step value="step2">
+        Step 2 content
+      </MultiStepPopover.Step>
+      <MultiStepPopover.Step value="step3">
+        Step 3 content
+      </MultiStepPopover.Step>
```

Comment:
I like the API - feels very Mantine-y. This kinda emulates the slot pattern you see in Vue/Svelte, which is nice.

Reply:
I'm kinda falling in love with the namespace component setup 😄